backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/modelvalidator/M_ReceiptSchedule.java

```diff
@@ -175,4 +175,15 @@ public void updateCanBeExportedAfter(@NonNull final I_M_ReceiptSchedule sched)
 	{
 		receiptScheduleBL.updateCanBeExportedFrom(sched);
 	}
+
+
+	@ModelChange(timings = {ModelValidator.TYPE_BEFORE_CHANGE}, ifColumnsChanged = {I_M_ReceiptSchedule.COLUMNNAME_MovementDate})
+	public void resetDatePromised_Override(@NonNull final I_M_ReceiptSchedule schedule)
+	{
+		//reset DatePromised_Override if MovementDate was set
+		if (schedule.getMovementDate() != null)
```

Comment:
is the movement date set manually by user or it's set by some API?
in case is set by an api, why not introducing a method which sets the MovementDate and resets the DatePromised_Override.